samples/bluetooth/mesh/light_ctrl/src/model_handler.c

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/mesh/models.h>
+#include <dk_buttons_and_leds.h>
+#include "model_handler.h"
+#include "lc_pwm_led.h"
+
+#define PWM_SCALING_FACTOR 512
+
+struct lightness_ctx {
+	struct bt_mesh_lightness_srv lightness_srv;
+	struct k_timer trans_timer;
+	struct k_delayed_work per_work;
+	u16_t target_lvl;
+	u16_t current_lvl;
+	u32_t time_per;
+};
+
+/** Configuration server definition */
+static struct bt_mesh_cfg_srv cfg_srv = {
+	.relay = IS_ENABLED(CONFIG_BT_MESH_RELAY),
+	.beacon = BT_MESH_BEACON_ENABLED,
+	.frnd = IS_ENABLED(CONFIG_BT_MESH_FRIEND),
+	.gatt_proxy = IS_ENABLED(CONFIG_BT_MESH_GATT_PROXY),
+	.default_ttl = 7,
+
+	/* 3 transmissions with 20ms interval */
+	.net_transmit = BT_MESH_TRANSMIT(2, 20),
+	.relay_retransmit = BT_MESH_TRANSMIT(2, 20),
+};
+
+/* Set up a repeating delayed work to blink the DK's LEDs when attention is
+ * requested.
+ */
+static struct k_delayed_work attention_blink_work;
+
+static void attention_blink(struct k_work *work)
+{
+	static int idx;
+	const u8_t pattern[] = {
+		BIT(0) | BIT(1),
+		BIT(1) | BIT(2),
+		BIT(2) | BIT(3),
+		BIT(3) | BIT(0),
+	};
+	dk_set_leds(pattern[idx++ % ARRAY_SIZE(pattern)]);
+	k_delayed_work_submit(&attention_blink_work, K_MSEC(30));
+}
+
+static void attention_on(struct bt_mesh_model *mod)
+{
+	k_delayed_work_submit(&attention_blink_work, K_NO_WAIT);
+}
+
+static void attention_off(struct bt_mesh_model *mod)
+{
+	k_delayed_work_cancel(&attention_blink_work);
+	dk_set_leds(DK_NO_LEDS_MSK);
+}
+
+static const struct bt_mesh_health_srv_cb health_srv_cb = {
+	.attn_on = attention_on,
+	.attn_off = attention_off,
+};
+
+static struct bt_mesh_health_srv health_srv = {
+	.cb = &health_srv_cb,
+};
+
+BT_MESH_HEALTH_PUB_DEFINE(health_pub, 0);
+
+static void start_new_light_trans(
+	const struct bt_mesh_lightness_set *set,
+	struct lightness_ctx *ctx)
+{
+	u32_t step_cnt = abs(set->lvl - ctx->current_lvl) / PWM_SCALING_FACTOR;
+
+	k_delayed_work_cancel(&ctx->per_work);
+	k_timer_stop(&ctx->trans_timer);
+	ctx->target_lvl = set->lvl;
+	ctx->time_per = (step_cnt ? set->transition->time / step_cnt : 0);
+	k_delayed_work_submit(&ctx->per_work, K_MSEC(set->transition->delay));
+	k_timer_start(&ctx->trans_timer,
+		K_MSEC(set->transition->time + set->transition->delay),
+		K_NO_WAIT);
```

Comment:
You don't really need this timer:

`remaining_time = abs(ctx->target_lvl - ctx->current_lvl) / (PWM_SCALING_FACTOR * ctx->time_per)`

Reply:
Fixed it in a simmilar manner